Src/HALALMock/Services/DigitalOutputService/DigitalOutputService.cpp

```diff
@@ -8,14 +8,19 @@
 
 #include "DigitalOutputService/DigitalOutputService.hpp"
 #include "ErrorHandler/ErrorHandler.hpp"
-#include "HALALMock/Services/SharedMemory/SharedMemory.hpp"
+
 
 
 uint8_t DigitalOutputService::id_counter = 0;
 map<uint8_t,Pin> DigitalOutputService::service_ids = {};
 
 uint8_t DigitalOutputService::inscribe(Pin& pin){
-        Pin::inscribe(pin, OUTPUT);
+        EmulatedPin &pin_data = SharedMemory::get_pin(pin);
+		if(pin_data.type == PinType::NOT_USED){
+			pin_data.type = PinType::DigitalOutput;
+			bool *output = &(pin_data.PinData.DigitalOutput.state);
+		}
+		*output = PinState::OFF;
```

Comment:
since you cannot store this variable in the class I think you could directly do this>
```suggestion
		}
		pin_data.PinData.DigitalOutput.state = PinState::OFF;
```